code/GatewayFieldsFactory.php

```diff
@@ -430,28 +377,55 @@ public function getGatewayFieldName($defaultName, $gateway = null)
     }
 
     /**
-     * DRY helper to fetch the rename configuration
-     * @return array|bool|\scalar
+     * Fetch the GatewayFieldsFactory configuration
+     * @return \Config_ForClass
      */
-    public function getRenameConfig()
+    public static function config()
     {
-        $renameMap = \Config::inst()->get('GatewayFieldsFactory', 'rename');
-
-        if (!$renameMap || empty($renameMap)) {
-            return false;
-        }
-
-        return $renameMap;
-
+        return \Config::inst()->forClass('GatewayFieldsFactory');
     }
 
     /**
      * Builds the rename map which is used as a lookup table for normalizeFieldData()
      * @return void
      */
-    public function buildRenameMap() {
+    private function buildRenameMap() {
+
+        $renameMap = $this->config()->rename;
+
+        if (!$renameMap) {
+            return;
+        }
+
+        $prefix = $this->getGatewayFieldName('prefix');
+        // if a prefix has already been defined in the gateway namespace, continue using it
+        // if not, and we have a global prefix, use that.
+        // if not, $prefix becomes an empty string
+        $prefix = ($prefix) ? $prefix : ($this->getGlobalFieldName('prefix') ?: '');
+
+
         foreach (static::$whitelist as $defaultName) {
```

Comment:
Sorry, I obviously missed this one in the previous review. It should be `$this->config()->whitelist` instead of directly accessing the static member. That enables users to update the whitelist via config system…